### ISSUE TITLE
oids counter starts at zero; spec says it should be random

### ISSUE THREAD

**Issue by dhasenan** [nim-lang/Nim#2796]
http://docs.mongodb.org/manual/reference/object-id/ says that the 'counter' field of an object ID should start at a random value. In oids.nim, it starts at 0.

I have a patch for this, and I'm using this issue to get approval from my workplace to commit to Nim.

**Comment by flaviut** [comment]
If your proposed fix uses `math.random`, it won't be a useful change. It'd essentially expand the clock value to 7 byes, 3 of which would be redundant.

**Comment by dhasenan** [comment]
That's only if the counter were re-randomized each iteration.

**Comment by dhasenan** [comment]
That would only be true if the counter were re-randomized each time the clock changed, and even then not so much.

**Comment by dom96** [comment]
What's the status of this?

**Comment by andreaferretti** [comment]
The linked documentation reads

```
The 12-byte ObjectId value consists of:

    a 4-byte value representing the seconds since the Unix epoch,
    a 3-byte machine identifier,
    a 2-byte process id, and
    a 3-byte counter, starting with a random value.
```

but the implementation in `oids.nim` is

```nim
type
  Oid* = object ## an OID
    time: int32  ##
    fuzz: int32  ##
    count: int32 ##
```

where fuzz is random but fixed. This does not seem to be too compatible.

Either

* we update the `oids` module, or
* we declare and document that nim oids are unique but not mongo compatible, or
* we remove `oids` from the standard library

**Comment by Araq** [comment]
The module was written for Mongodb though and I ported the C code from it. No idea why we can't simply fix this and call it a day.

**Comment by narimiran** [comment]
Closed by #10248

**Comment by timotheecour** [comment]
> The module was written for Mongodb though and I ported the C code from it (it looks like Mongo later changed its implementation). No idea why we can't simply fix this and call it a day.

that's tracked here: [genOid has low entropy (creates lots of 0 bits) · Issue #9741 · nim-lang/Nim](https://github.com/nim-lang/Nim/issues/9741)
